package/com.unity.formats.usd/Runtime/Scripts/IO/Geometry/MeshImporter.cs

```diff
@@ -740,36 +909,41 @@ public class GeometrySubsets {
     /// An array of size > 0 on succes, an array of size 0 on failure, or null if the given object
     /// is not of the desired type T.
     /// </returns>
-    private static T[] TryGetUVSet<T>(object uv,
-                                      int[] uvIndices,
-                                      int[] faceVertexCounts,
-                                      int[] faceVertexIndices,
-                                      int vertexCount,
-                                      GameObject go) {
-      if (uv.GetType() != typeof(T[])) {
+    private static T[] TryGetUVSet<T>(
+      Primvar<object> uv,
+      int[] faceVertexCounts,
+      int[] faceVertexIndices,
+      int vertexCount,
+      GameObject go)
+    {
+      if (uv.value.GetType() != typeof(T[]))
+      {
         return null;
       }
 
-      var uvVec = (T[])uv;
+      var uvVec = (T[]) uv.value;
 
-      if (uvVec.Length == 0) {
+      if (uvVec.Length == 0)
+      {
         return uvVec;
       }
 
       // Unroll UV indices if specified.
-      if (uvIndices != null && uvIndices.Length > 0) {
-        var newUvs = new T[uvIndices.Length];
-        for (int i = 0; i < uvIndices.Length; i++) {
-          newUvs[i] = uvVec[uvIndices[i]];
+      if (uv.indices != null && uv.indices.Length > 0)
+      {
+        var newUvs = new T[uv.indices.Length];
+        for (int i = 0; i < uv.indices.Length; i++)
+        {
+          newUvs[i] = uvVec[uv.indices[i]];
         }
         uvVec = newUvs;
       }
 
       // If there are more UVs than verts, the UVs must be face varying, e.g. each vertex for
       // each face has a unique UV value. These values must be collapsed such that verts shared
       // between faces also share a single UV value.
-      if (uvVec.Length > vertexCount) {
-        uvVec = UnrollFaceVarying(vertexCount, uvVec, faceVertexCounts, faceVertexIndices);
+      if (uvVec.Length != vertexCount) {
+//        uvVec = UnrollFaceVarying(vertexCount, uvVec, faceVertexCounts, faceVertexIndices);
```

Comment:
dead comment?

Reply:
Actually I'm missing some refactoring here as well... I did nothing except commenting the call to the function ^^